packages/neo-one-server-plugin-project/src/initCommand.ts

```diff
@@ -7,37 +7,54 @@ import jsonFile from 'json-file-plus';
 import * as path from 'path';
 import { loadProjectConfig } from './utils';
 
+interface InitDataFileOpts {
+  readonly folder: string;
+  readonly name: string;
+  // tslint:disable-next-line:no-any
+  readonly data: any;
+  readonly cli: InteractiveCLIArgs['cli'];
+}
+export const initDataFile = async ({ folder, name, data, cli }: InitDataFileOpts) => {
+  const filePath = path.resolve(folder, name);
+  await fs.ensureDir(folder);
+  await fs.writeFile(
+    filePath,
+    JSON.stringify(
+      data,
+      // tslint:disable-next-line:no-null-keyword
+      null,
+      2,
+    ),
+  );
+  cli.print(`Created ${name} at ${filePath}.`);
+};
+
 export const initCommand = ({ cli }: InteractiveCLIArgs) =>
   cli.vorpal.command('init', 'Initialize neo-one in the current directory.').action(async () => {
     const cwd = process.cwd();
     const projectConfig = await loadProjectConfig(cwd);
     const tsconfigPath = path.resolve(projectConfig.paths.contracts, 'tsconfig.json');
     const contractsPath = path.dirname(tsconfigPath);
-
-    await fs.ensureDir(contractsPath);
-    await fs.writeFile(
-      tsconfigPath,
-      JSON.stringify(
-        {
-          compilerOptions: {
-            ...COMPILER_OPTIONS,
-            target: 'esnext',
-            module: 'esnext',
-            moduleResolution: 'node',
-            jsx: 'react',
-            plugins: [
-              {
-                name: '@neo-one/smart-contract-typescript-plugin',
-              },
-            ],
+    const tsData = {
+      compilerOptions: {
+        ...COMPILER_OPTIONS,
+        target: 'esnext',
+        module: 'esnext',
+        moduleResolution: 'node',
+        jsx: 'react',
+        plugins: [
+          {
+            name: '@neo-one/smart-contract-typescript-plugin',
           },
-        },
-        // tslint:disable-next-line:no-null-keyword
-        null,
-        2,
-      ),
-    );
-    cli.print(`Created tsconfig.json at ${tsconfigPath}.`);
+        ],
+      },
+    };
+    const rcData = { codegen: projectConfig.codegen };
```

Comment:
I'm not sure I quite understand what this is doing - we load the project config from the cwd and then write it back out?

Reply:
Pulling in defaults.  If something is defined in the existing `.onerc`, it is used, otherwise use defaults.